internal/web/ssehub.go

```diff
@@ -0,0 +1,91 @@
+package web
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+)
+
+// SSEHub maintains the set of active clients and broadcasts messages to the clients.
+type SSEHub struct {
+	// Inbound messages from the clients.
+	Broadcast chan *sse
+
+	// Register requests from the clients.
+	Register chan *sseClient
+
+	// Unregister requests from clients.
+	Unregister chan *sseClient
+
+	// Registered clients.
+	clients map[*sseClient]bool
+}
+
+type sse struct {
+	event string
+	data  string
+}
+
+func (evt *sse) String() string {
+	return fmt.Sprintf("event: %s\ndata: %s\n\n", evt.event, strings.ReplaceAll(evt.data, "\n", ""))
+}
+
+type sseClient struct {
+	send chan *sse
+}
+
+// NewHub creates new SSEHub
+func NewHub() *SSEHub {
+	return &SSEHub{
+		Broadcast:  make(chan *sse),
+		Register:   make(chan *sseClient),
+		Unregister: make(chan *sseClient),
+		clients:    make(map[*sseClient]bool),
```

Comment:
Does the `bool` do anything here? If not, you could use a `map[*sseClient]struct{}`, which is a little bit more efficient.